SimTracker/TrackTriggerAssociation/plugins/TTStubAssociator.cc

```diff
@@ -205,34 +193,29 @@ void TTStubAssociator<Ref_Phase2TrackerDigi_>::produce(edm::Event& iEvent, const
 
     /// Clean the only map that needs cleaning
     /// Prepare the output map wrt TrackingParticle
-    typename std::map<edm::Ptr<TrackingParticle>,
-                      std::vector<edm::Ref<edmNew::DetSetVector<TTStub<Ref_Phase2TrackerDigi_>>,
-                                           TTStub<Ref_Phase2TrackerDigi_>>>>::iterator iterMapToClean;
-    for (iterMapToClean = trackingParticleToStubVectorMap.begin();
-         iterMapToClean != trackingParticleToStubVectorMap.end();
-         ++iterMapToClean) {
+    typename std::map<TrackingParticlePtr, std::vector<TTStubRef>>::iterator iterMapToClean;
+    for (auto& p : trackingParticleToStubVectorMap) {
       /// Get the vector of references to TTStub
-      std::vector<edm::Ref<edmNew::DetSetVector<TTStub<Ref_Phase2TrackerDigi_>>, TTStub<Ref_Phase2TrackerDigi_>>>
-          tempVector = iterMapToClean->second;
+      std::vector<TTStubRef> tempVector = p.second;
```

Comment:
avoid copy and reassignment: `std::vector<TTStubRef>& tempVector = p.second;`
(then delete line `p.second = tempVector;` below)

Reply:
done